### PR TITLE
2202 deleted dq rules

### PR DESCRIPTION
**Pull request by adrian-lara**
#### Any background context you want to provide?
DQ Rules could inadvertently be deleted. See tagged issue for more details.

#### What's this PR do?
Cherry-pick changes from 2.7.0-patch1.
- Performs validity checks on incoming DQ Rule updates _before_ the teardown and rebuild of DQ Rules (a pattern that is slated to be removed).
- Skips over DQ Rules that could not be recreated instead of stopping the recreation process entirely.

#### How should this be manually tested?
Try to save a "Valid Data" Rule without a label. See that nothing gets saved, and we get the same 400 we got from before. Refresh the page and see that nothing was lost either.

Try to save a Rule as Must Contain or Must Not Contain while keeping the text field blank. See that nothing gets saved, and we get the same 400 we got from before. Refresh the page and see that nothing was lost either.

For code review, commits can be reviewed in the order shown.

#### What are the relevant tickets?
#2202

### CODE REVIEW

File: seed/views/data_quality.py [hunk 1]
```diff
@@ -405,29 +428,27 @@ def save_data_quality_rules(self, request, pk=None):
...
-                    'status': 'error',
-                    'message': e,
-                }, status=status.HTTP_400_BAD_REQUEST)
-
-        return self.data_quality_rules(request)
+            with transaction.atomic():
+                try:
+                    dq.add_rule(rule)
+                except Exception as e:
+                    error_messages.add('Rule could not be recreated: ' + str(e))
```
Comment (lainsworth8801) [inline seed/views/data_quality.py:441]: Verified:
Screen Shot 2020-06-01 at 5 19 50 PM

**Comment by adrian-lara** [comment]
> On the UI front it still shows just the first error captured:
> Screen Shot 2020-06-02 at 8 38 00 PM
> 
> There might be some additional changes required in menu_controller.js to show all errors on the admin page.

I can get the multiple error messages, though it shows up in one line (vs the intended multiple lines). 
Screen Shot 2020-06-03 at 5 42 46 PM

There does seem to be an issue with throwing errors on subsequent "Save Changes". The text doesn't seem to update at all on the 2nd+ save attempt unless you refresh the page first. I'm not sure if the `$emit('app_error')` pattern was meant for this type of error handling. Either way, I think we'll eventually want to transition away from using that pattern and possibly use Notifications, but that effort isn't in scope here.

Given the primary goal of this PR is to temporarily address the inadvertently deleted DQ Rules, we snowballed a little when included handling multiple errors on the frontend, so I'd like to avoid snowballing this PR anymore. 

Let me know what you think.

**Comment by lainsworth8801** [comment]
@adrian-lara I think it makes sense. Let’s at least keep #2207 open for tracking purpose?

**Comment by lainsworth8801** [comment]
Yes I was able to see both errors with saving changes once. Like you said, consistently saving changes in between errors only display the first error. I will update #2207 with this info thanks!

**Pull request merged**